raft-tech/TANF-app
# Space Fill Checks for T1's, T4's, M1, and M4 Records

## Summary of Changes
- Updated preparsing on T1, T4, M1, M4, Tribal T1, and Tribal T4 to support minimum length requirements to parse
- Updated the case aggregates `rejected` query to not count multiplicities per record
- Added test to cover the `rejected` count
Pull request closes #2826 

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
2. Submit a datafile with any of the above mentioned records that do and don't meet length requirements
3. Verify you do or don't see the appropriate error messages.
4. Submit `aggregates_rejected.txt` and verify you see 3 pre-check parser errors, but only 1 rejected record. Beforehand, you would have seen the count of rejected records as 3.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] error counts not duplicated for rejected
+ [x] TANF T1 records are parsed if 117<= length <= 156
+ [x] Tribal TANF and TANF T4 records are parsed if 36 <= length <= 71
+ [x] Tribal TANF T1 records are parsed if 117<= length <= 122
+ [x] SSP M1 records are parsed if 113 <= length <= 150
+ [x] SSP M4 records are parsed if 36 <= length <= 66

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
